pyramid_zipkin/request_helper.py

```diff
@@ -19,18 +20,31 @@ def get_trace_id(request):
     :returns: a 64-bit hex string
     """
     if 'X-B3-TraceId' in request.headers:
+        trace_id = _convert_signed_hex(request.headers['X-B3-TraceId'])
         # Tolerates 128 bit X-B3-TraceId by reading the right-most 16 hex
         # characters (as opposed to overflowing a U64 and starting a new trace).
-        trace_id = request.headers['X-B3-TraceId'][-16:]
+        trace_id = trace_id[-16:]
     elif 'zipkin.trace_id_generator' in request.registry.settings:
-        trace_id = request.registry.settings[
-            'zipkin.trace_id_generator'](request)
+        trace_id = _convert_signed_hex(request.registry.settings[
+            'zipkin.trace_id_generator'](request))
     else:
         trace_id = generate_random_64bit_string()
 
     return trace_id
 
 
+def _convert_signed_hex(s):
+    """Takes a signed hex string that begins with '0x' and converts it to
```

Comment:
note that 0x prefix is definitely not zipkin's id encoding, neither is negative hex :P I presume this is because there's some old code out there emitting this?